### ISSUE TITLE
Texture scaling breaks some textures from recent color conversion changes.

### ISSUE THREAD

**Issue by LunaMoo** [hrydgard/ppsspp#7364]
The issue I got while testing the hack in #6192 was actually caused by some of the recent changes:]. It actually causes alot of broken stuff in some games, like for example Parasite Eve 3 which I was also testing against the hack:
![ules01513_00001](https://cloud.githubusercontent.com/assets/5485237/5860236/4efdc0aa-a260-11e4-9aa3-c107a9e05fdd.jpg)

Happens with texture scaling with both backends for texture type 5551(the one from GE2 transitioning screen which I posted in the bloom issue):
![ge2 texture](https://cloud.githubusercontent.com/assets/5485237/5860781/25663df4-a264-11e4-95e5-b1af83a48543.jpg)

However that weirdness in PE3 happens only in OGL, shadows are:
![pe3](https://cloud.githubusercontent.com/assets/5485237/5860432/93cf12fa-a261-11e4-9f1c-8b4aec3ca378.jpg)

and snowflakes:
![pe3-1](https://cloud.githubusercontent.com/assets/5485237/5860511/08f0bb88-a262-11e4-9eca-db75b04c9035.jpg)

Edited the title, since the clut 4 / 4444 texture type issue actually happened in v0.9.9.1-1580-gf1b57da	as well, while the texture type 5551 issue is just from v0.9.9.1-1582-g08eaa6e which was supposed to be a fix for it?

 Anyway to sum it up since it might be hard to notice between screenshots:
- 5551 texture type - broken with texture scaling in both backends from v0.9.9.1-1582-g08eaa6e,
- clut 4 / 4444 texture type - broken with texture scaling only in OGL from one of the earlier commits of that cleanup,
- last working version without neither of the issues v0.9.9.1-1576-g4100163.

Also just to mention this http://forums.ppsspp.org/showthread.php?tid=14430 is same issue as with PE3, same texture type.

**Comment by LunaMoo** [comment]
All the problematic commits were reverted, so closing:3.

**Comment by hrydgard** [comment]
Will do it again later, more carefully :) Then we will be in a good position to speed these up, although I don't think they're usually big bottlenecks anyway..

**Comment by unknownbrackets** [comment]
Well, optimizations can matter in games with a lot of texture turnover (changes) or for load times in some cases.

-[Unknown]

**Comment by hrydgard** [comment]
Yeah it will help a bit and probably noticeably in some cases, just meant it's not likely to revolutionize speed :)
